tests/ssl-opt.sh

```diff
@@ -1401,6 +1401,22 @@ run_test    "Renegotiation: client-initiated" \
             -s "=> renegotiate" \
             -S "write hello request"
 
+# Checks if the server parses the SigAlgs Extension. Valid only if an MD
+# algorithm stronger than SHA1 is enabled in config.h
```

Comment:
We have `requires_config_enabled` for that purpose, which however only accepts only a single configuration flag.

Reply:
@yanesca Could you extend the comment to explain a bit how the test is supposed to work? It took me a bit to understand.

Reply:
We could use something like 
```
# skip next test if none of the flags is enabled in config.h
requires_config_enabled_or() {
    SKIP_NEXT="YES"
    for var in "$@"
    do
        if grep "^#define $var" $CONFIG_H > /dev/null; then
            SKIP_NEXT="NO"
        fi
    done
}
```
and prefix the test with `requires_config_enabled_or MBEDTLS_SHA256_C MBEDTLS_SHA512_C`.

Reply:
I think this is a separate issue and bigger on its own than the current one. This isn't the first and only test where this feature would be necessary to detect compilation flags properly.

Fixing this problem should involve not just adding the feature to the script, but going through all the tests and flagging them properly.

In summary: I think this is out of scope and would prefer to leave it like it is.

Reply:
@yanesca Fair enough. I'd still prefer to just add it for this test only so that it's easier in the future to adapt the other tests, but you're right that this critical PR should probably be kept as small as possible.

Reply:
@hanno-arm It is not just because this PR is critical, in general I prefer if a PR (not just mine) only does what it needs to do and doesn't add a lot of ad hoc changes.

Reply:
Is there an issue tracking this problem generally? If not, could you please file one appropriately?

Reply:
Yes, this issue is already being tracked internally.